speaker/layer2_controller.go

```diff
@@ -37,6 +37,16 @@ func (c *layer2Controller) SetConfig(log.Logger, *config.Config) error {
 	return nil
 }
 
+func activeNodes(mList *memberlist.Memberlist) []string {
+	var activeNodes []string
+	if mList != nil {
+		for _, n := range mList.Members() {
+		    activeNodes = append(activeNodes, n.Name)
+		}
+	}
+    return activeNodes
+}
```

Comment:
The code below in `usableNodes()` already builds up an `activeNodes` map.  Maybe just update that to return both the active and usable nodes?  You can return more than one thing in golang.